async-csharp/AsynchronousProgrammingPatterns/AsynchronousProgrammingPatterns/Providers/EapUserProvider.cs

```diff
@@ -0,0 +1,62 @@
+﻿using AsynchronousProgrammingPatterns.Services;
+using AsynchronousProgrammingPatterns.Services.Models;
+using System.ComponentModel;
+
+namespace AsynchronousProgrammingPatterns.Providers;
+
+public class EapUserProvider
+{
+    private readonly SendOrPostCallback _operationFinished;
+    private readonly UserService _userService;
+
+    public EapUserProvider()
+    {
+        _operationFinished = ProcessOperationFinished;
+        _userService = new UserService();
+    }
+
+    public User GetUser(int userId) => _userService.GetUser(userId);
+
+    public event EventHandler<GetUserCompletedEventArgs>
+      GetUserCompleted;
+
+    public void GetUserAsync(int userId, object userState)
```

Comment:
@akarshvijayan I can see the `userState` is always passed as null, so do we actually need this param? I believe it's optional to provide this param. I've mentioned this in the review doc as well, additional params should be part of an overloaded method.

Reply:
Created overload.